ludwig/trainers/trainer.py

```diff
@@ -214,6 +215,24 @@ def prepare(self):
 
         # We may need to replace the embedding layer when using 8-bit optimizers from bitsandbytes.
         update_embedding_layer(self.compiled_model, self.config)
+
+        # Enable gradient checkpointing if configured
+        if self.config.enable_gradient_checkpointing:
+            # TODO(Arnav): Add support for gradient checkpointing in the compiled model
+            # when the model is an ECD model using torch.utils.checkpoint (torch.utils.checkpoint.sequential())
+            if not isinstance(self.compiled_model, LLM):
+                logger.warning("Gradient checkpointing is currently only supported for model_type: llm. Skipping...")
+            elif not hasattr(self.compiled_model, "model") and not hasattr(
+                self.compiled_model.model, "gradient_checkpointing_enable"
+            ):
```

Comment:
Is it worth distinguishing two different warnings between these two cases?

Reply:
I don't think so, since self.compiled_model.model only exists for LLM model types, otherwise its just torch.nn modules with the actual architecture beneath the surface, so I would treat these the same